### PR TITLE
support cert/key reload without server restart

### PR DESCRIPTION
**Pull request by jiuker**
support cert/key reload without server restart
fix https://github.com/minio/sidekick/issues/106
how to test:
```shell
PS C:\workspace\go\src\sidekick\dist> ./certgen -host "192.168.2.6,127.0.0.1,localhost"             
Created a new certificate 'public.crt', 'private.key' valid for the following names 📜
 - "192.168.2.6"
 - "127.0.0.1"
 - "localhost"
PS C:\workspace\go\src\sidekick\dist> mc alias set sk https://192.168.2.6:8090 minioadmin minioadmin
Fingerprint of sk public key: 2385a69ce94fb9f70e9a53fd5a91310c41c93a394e9ff659877310e20ce7d57b
Confirm public key y/N: y
mc.exe: <ERROR> Unable to initialize new alias from the provided credentials. Get "https://192.168.2.6:8090": tls: failed to verify certificate: x509: certificate signed by unknown authority.
PS C:\workspace\go\src\sidekick\dist> ./certgen -host "127.0.0.1,localhost"                         
Created a new certificate 'public.crt', 'private.key' valid for the following names 📜
 - "127.0.0.1"
 - "localhost"
PS C:\workspace\go\src\sidekick\dist> mc alias set sk https://192.168.2.6:8090 minioadmin minioadmin
mc.exe: <ERROR> Unable to initialize new alias from the provided credentials. Get "https://192.168.2.6:8090": tls: failed to verify certificate: x509: certificate is valid for 127.0.0.1, not 192.168.2.6.
PS C:\workspace\go\src\sidekick\dist> 
```

### CODE REVIEW

**Review by ramondeklein — APPROVED** [review]

**Review by klauspost — APPROVED** [review]

File: main.go [hunk 1]
```diff
@@ -910,7 +911,28 @@ func sidekickMain(ctx *cli.Context) {
...
 	router.PathPrefix(slashSeparator).Handler(m)
 	if ctx.String("cert") != "" && ctx.String("key") != "" {
-		if err := http.ListenAndServeTLS(addr, ctx.String("cert"), ctx.String("key"), router); err != nil {
+		manager, err := certs.NewManager(context.Background(), ctx.String("cert"), ctx.String("key"), tls.LoadX509KeyPair)
+		if err != nil {
+			console.Fatalln(err)
+		}
+		tlsConfig := &tls.Config{
+			PreferServerCipherSuites: true,
+			NextProtos:               []string{"h2", "http/1.1"},
```
Comment (klauspost) [inline main.go:920]: Do we really prefer http/2?

Comment (ramondeklein) [reply main.go:920]: I was asking that myself too, but I found exactly the same code in the Minio repo, so I guess it was copied from there.

Comment (ramondeklein) [reply main.go:920]: Now I want to refer to `cmd/utils.go:952` again, I see the order is different...

Comment (jiuker) [reply main.go:920]: Yesh. From console with 
```
			// Use modern tls mode https://wiki.mozilla.org/Security/Server_Side_TLS#Modern_compatibility
			NextProtos: []string{"h2", "http/1.1"},
```
But I check the go code here. Just use strSliceContains here. I think order is ok. @ramondeklein @klauspost 
@harshavardhana I don't know too.

Comment (harshavardhana) [reply main.go:920]: on MinIO we prefer http/1.1 first sidekick should also have `http/1.1` first. http2 is great but Go's implementation is shoddy and riddled with bugs.

**Review by harshavardhana — APPROVED** [review]

File: main.go [hunk 2]
```diff
@@ -43,6 +43,7 @@ import (
 
 	"github.com/minio/cli"
 	"github.com/minio/dnscache"
+	"github.com/minio/pkg/v2/certs"
 	"github.com/minio/pkg/v2/console"
 	"github.com/minio/pkg/v2/ellipses"
```
Comment (harshavardhana) [inline main.go:48]:
```suggestion
	"github.com/minio/pkg/v3/certs"
	"github.com/minio/pkg/v3/console"
	"github.com/minio/pkg/v3/ellipses"
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**